collector/benchmarks/README.md

```diff
@@ -246,3 +246,51 @@ Rust code being written today.
 - Compare the benchmarking time of the two versions.
 - When adding the new version, for `perf-config.json` and the `N-*.patch`
   files, use the corresponding files for the old version as a starting point.
+
+# Benchmark update policy
+
+rustc-perf is a "living benchmark suite" that is regularly changed. Some
+benchmarks in rustc-perf are verbatim copies of third-party crates. We
+periodically do a mass update of these benchmarks.
+
+Benefits of this approach:
+- We ensure we are measuring compilation of the crates most people are using.
+  This is most relevant for popular crates.
+- We get coverage of newer language features.
+
+Costs of this approach:
+- It takes time and effort.
+- We lose some data continuity.
+  - But the stable set of benchmarks used for the dashboard are not affected,
+    and they provide the greatest continuity.
+- If the code hasn't changed much, it won't have much effect.
+
+To balance these requires choosing a good refresh period. A period of three
+years is reasonable:
+- It's not too frequent (as 1 year would be, for example) while avoiding some
+  sub-optimal occurrences from the past (e.g. in 2022 numerous crates that were
+  more than five years old were updated).
+- It aligns with the Rust edition cycle.
```

Comment:
I'm not sure I follow how edition cycle is relevant since editions are really just surface level and don't really correspond to user's experience of compiler perf. While *some* new features will be tied to an edition, the vast majority are not.

Reply:
I suppose that it's a bit difficult to state rules like "after a major feature, try to add a benchmark that uses it", so tying the update cycle to editions at least gives some frequent cycle that is easily estimated and at the same time will likely correspond to new features and/or code patterns.

Reply:
The choice of time period here is definitely arbitrary. Alignment with the edition cycle isn't only moderately compelling, but it was more compelling than any other reason I could think of for any other time period. Can you think of a reason for a different time period?

Reply:
No, I cannot. I think I'd feel more comfortable if we were just honest in the docs about this being arbitrary.

Reply:
I think it's actually worth noting here that even though 3 years seems OK, I think we should intentionally "misalign" with the cycle used for editions. We've already somewhat done that (2021 edition was ~December, now is ~May), but maybe pushing that further apart would be a good idea. I think trying to do too many things at the same time is a recipe for them getting done poorly or not at all :)

Reply:
There's text to this effect lower down: "The update should occur roughly six months after the edition hits a stable Rust release, to give crate authors a decent amount of time to update to the new edition."